tests/integration/helpers/helpers.py

```diff
@@ -146,3 +147,7 @@ def ec_sign(
         int.to_bytes(signature.r, 32, "big"),
         int.to_bytes(signature.s, 32, "big"),
     )
+
+async def mine_block(provider: Web3, timestamp: int):
+    response = await provider.make_request("evm_mine", [timestamp])
```

Comment:
this will not work as we are not using `web3` related provider, you need to update the `starknet` fixture; you can take inspiration from what's done in `1_startknet.py` fixture:

```
    starknet.state.state.update_block_info(
        BlockInfo.create_for_testing(
            block_number=current_block_number, block_timestamp=current_block_timestamp
        )
    )
```

to summarize, you need to write a `@pytest.fixture` that would update the starknet state as in the previous example